File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/RuleFailure/InvalidSyntax.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace TestCasesForRuleFailure

    Public Sub Method2(i As Integer, j As Integer)
    End Sub

    Public Class DuplicatedNotImplementedInterfaces
        Implements IList
        Implements IList

    End Class

    Class InvalidSyntax

        Public Sub Method(i As Integer, j As Integer,)
            Dim x =
            For i as integer = 0 to 5
        End Sub

        Public Sub InvalidChars()
            ;
            {
            }
        End Sub

        Private Class C
            Public New
        End Class

    End Class

    Class

        i as integer

    End Class

    Public Class NoSubName

        Public Sub
        End Sub

    End Class

End Namespace

File: analyzers/tests/SonarAnalyzer.Test/TestCases/RuleFailure/InvalidSyntax.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text
using System.Threading.Tasks;

namespace SonarAnalyzer.Test.TestCasesForRuleFailure
{
    public void Method2(int i, int j) { }

    class DuplicatedInterfaces: IList, IList {}

    class InvalidSyntax
    {
        ;
        public void Method(int i, int j,)
        {
            var x = 6
            for (int i = 0; i < 5; i++)
            }

        private class C
        {
            public C()
        }
    }

    class
    {
        int i;
        public override int GetHashCode()
        {
            int? v = null;
            if (v > missing)
            { }

            return i; // we don't report on this
        }
    }
}
